stdlib/public/Distributed/DistributedActor.swift

```diff
@@ -75,6 +75,17 @@ public protocol DistributedActor:
     /// Conformance to this requirement is synthesized automatically for any
     /// `distributed actor` declaration.
     nonisolated var id: AnyActorIdentity { get }
+
+
+    /// Executes the passed 'body' only when the distributed actor is local instance.
+    ///
+    /// The `Self` passed to the the body closure is isolated, meaning that the
+    /// closure can be used to call non-distributed functions, or even access actor
+    /// state.
+    ///
+    /// When the actor is remote, the closure won't be executed and this function will return nil.
+    nonisolated func whenLocal<T>(_ body: @Sendable (isolated Self) async throws -> T)
```

Comment:
Why is this a requirement on the protocol? It seems like the implementation will always be the same, so it should be in a protocol extension, only.